zebra-consensus/src/error.rs

```diff
@@ -56,6 +57,13 @@ pub enum TransactionError {
     #[error("coinbase inputs MUST NOT exist in mempool")]
     CoinbaseInMempool,
 
+    #[error("transaction is locked until block height {}", _0.0)]
+    LockedUntilBlockHeight(block::Height),
```

Comment:
Required Consensus Rule Clarification: The lock still applies if the block height is equal to this height.

```suggestion
    #[error("transaction is locked until after block height {}", _0.0)]
    LockedUntilAfterBlockHeight(block::Height),
```